src/nnfusion/engine/pass/graph/autodiff_pass.cpp

```diff
@@ -64,8 +65,17 @@ bool AutodiffPass::run_on_graph(std::shared_ptr<Graph>& graph)
         auto gnode = graph->get_outputs()[i];
         NNFUSION_CHECK(gnode->get_output_size() == 1);
         outputs_index.emplace_back(gnode, 0);
-        auto one_op =
-            std::make_shared<op::Constant>(element::f32, gnode->get_shape(), std::vector<float>{1});
+        std::shared_ptr<op::Constant> one_op;
+        if (backward_inputs != nullptr && i < backward_inputs->size())
+        {
+            one_op = std::make_shared<op::Constant>(
+                element::f32, gnode->get_shape(), backward_inputs->at(i));
+        }
+        else
+        {
+            one_op = std::make_shared<op::Constant>(
+                element::f32, gnode->get_shape(), std::vector<float>{1});
+        }
```

Comment:
🟢 [Option]

How about this:

auto vec = std::vector<float>{1};
if (...)
{
   vec = backward_inputs->at(i);
}

one_op = make_shared<>(...);

Reply:
Done.